src/key.rs

```diff
@@ -1189,6 +1267,14 @@ impl XOnlyPublicKey {
             err == 1
         }
     }
+
+    /// Returns the [`PublicKey`] for this [`XOnlyPublicKey`].
+    ///
+    /// This is equivalent to using [`PublicKey::from_xonly_and_parity(self, parity)`].
+    #[inline]
+    pub fn public_key(&self, parity: Parity) -> Result<PublicKey, Error> {
+        PublicKey::from_xonly(*self, parity)
```

Comment:
After the abox fix, this should not return `Result`. And I think we should have not a parity parameter here.